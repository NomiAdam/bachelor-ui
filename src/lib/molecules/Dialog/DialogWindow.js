import React from 'react';
import PropTypes from 'prop-types';
import Paper from '../../organisms/Paper/index';
import { Grid, GridCol } from '../../organisms/Grid/index';
import { DialogContent, DialogActions } from './components/index';

const DialogWindow = ({
    actionChildren, children, heading, width, height, backgroundColor,
}) => (
    <Paper width={ width } height={ height } backgroundColor={ backgroundColor }>
        <Grid gridGap="0" padding="0">
            <GridCol colXS={ 12 }>
                {heading}
            </GridCol>
            <GridCol colXS={ 12 }>
                <DialogContent>
                    {children}
                </DialogContent>
            </GridCol>
            <GridCol colXS={ 12 }>
                <DialogActions>
                    { actionChildren }
                </DialogActions>
            </GridCol>
        </Grid>
    </Paper>
);

DialogWindow.propTypes = {
    /**
     * Children nodes
     */
    children: PropTypes.any,
    /**
     * Function handler to render action children
     */
    actionChildren: PropTypes.any.isRequired,
    /**
     * Node rendered on heading
     */
    heading: PropTypes.any.isRequired,
    /**
   * string definition of dialog width
   */
    width: PropTypes.string,
    /**
   * string definition of dialog height
   */
    height: PropTypes.string,
    /**
   * String representation of outer backgroundColor css property
   */
    backgroundColor: PropTypes.string,
};

export default DialogWindow;
