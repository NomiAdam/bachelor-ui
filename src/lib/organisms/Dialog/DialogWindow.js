import React from 'react';
import PropTypes from 'prop-types';
import Paper from '../../molecules/Paper/index';
import { Grid, GridCol } from '../../molecules/Grid/index';
import { DialogContent, DialogActions } from './components/index';

const preventClick = (e) => {
    e.stopPropagation();
    e.preventDefault();
    return false;
};

const DialogWindow = ({
    actionChildren, children, heading, width, height, backgroundColor,
}) => (
    <Paper width={ width } height={ height } backgroundColor={ backgroundColor } handleClick={ preventClick }>
        <Grid gridGap="0" padding="0">
            <GridCol colXS={ 12 }>
                {heading}
            </GridCol>
            <GridCol colXS={ 12 }>
                <DialogContent>
                    {children}
                </DialogContent>
            </GridCol>
            <GridCol colXS={ 12 }>
                <DialogActions>
                    { actionChildren }
                </DialogActions>
            </GridCol>
        </Grid>
    </Paper>
);

DialogWindow.propTypes = {
    /**
     * Children nodes
     */
    children: PropTypes.any,
    /**
     * Function handler to render action children
     */
    actionChildren: PropTypes.any.isRequired,
    /**
     * Node rendered on heading
     */
    heading: PropTypes.any.isRequired,
    /**
   * string definition of dialog width
   */
    width: PropTypes.string,
    /**
   * string definition of dialog height
   */
    height: PropTypes.string,
    /**
   * String representation of outer backgroundColor css property
   */
    backgroundColor: PropTypes.string,
};

export default DialogWindow;
