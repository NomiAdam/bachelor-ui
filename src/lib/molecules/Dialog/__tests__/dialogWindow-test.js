import React from 'react';
import DialogWindow from '../DialogWindow';
import { shallow } from 'enzyme';

const mockChildren = () => <h1>Mock</h1>;

describe('DialogWindow', () => {
    it('should be defined', () => {
        expect(DialogWindow).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<DialogWindow renderActionChildren={mockChildren} />)).toMatchSnapshot();
    });
    it('should match the snapshot with children', () => {
        expect(shallow(<DialogWindow renderActionChildren={mockChildren} isOpen>{ mockChildren() }</DialogWindow>)).toMatchSnapshot();
    });
});
