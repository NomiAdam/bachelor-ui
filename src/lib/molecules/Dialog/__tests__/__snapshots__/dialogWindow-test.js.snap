// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogWindow should match the snapshot 1`] = `
<Paper
  background="#ffffff"
  overflow="hidden"
  padding="10px"
  radius="0"
>
  <styled.div
    gridGap="0"
    padding="0"
  >
    <styled.div
      colXS={12}
      horizontalAlign="flex-start"
      verticalAlign="flex-start"
    />
    <styled.div
      colXS={12}
      horizontalAlign="flex-start"
      verticalAlign="flex-start"
    >
      <DialogContent />
    </styled.div>
    <styled.div
      colXS={12}
      horizontalAlign="flex-start"
      verticalAlign="flex-start"
    >
      <DialogActions />
    </styled.div>
  </styled.div>
</Paper>
`;

exports[`DialogWindow should match the snapshot with children 1`] = `
<Paper
  background="#ffffff"
  overflow="hidden"
  padding="10px"
  radius="0"
>
  <styled.div
    gridGap="0"
    padding="0"
  >
    <styled.div
      colXS={12}
      horizontalAlign="flex-start"
      verticalAlign="flex-start"
    />
    <styled.div
      colXS={12}
      horizontalAlign="flex-start"
      verticalAlign="flex-start"
    >
      <DialogContent>
        <h1>
          Mock
        </h1>
      </DialogContent>
    </styled.div>
    <styled.div
      colXS={12}
      horizontalAlign="flex-start"
      verticalAlign="flex-start"
    >
      <DialogActions />
    </styled.div>
  </styled.div>
</Paper>
`;
